src/interface/metadata.hpp

```diff
@@ -193,6 +196,108 @@ class Metadata {
   PARTHENON_INTERNAL_FOREACH_BUILTIN_FLAG
 #undef PARTHENON_INTERNAL_FOR_FLAG
 
+  // TODO(JMM): Kind of treating Metadata as a namespace here
+  using FlagVec = std::vector<MetadataFlag>;
+
+  // FlagCollection is a collection object that encapsulates the desire for
+  // target variables/swarms with:
+  // - At least ONE of the "unions" flags,
+  // - ALL of the "intersections" flags,
+  // - and NONE of the "exclusions" flags
+  class FlagCollection {
+   public:
+    FlagCollection() = default;
+    // Container_t should be a stdlib container type
+    // it needs at least the iterator first and last methods
+    template <template <class...> class Container_t, class... extra>
+    FlagCollection(const Container_t<MetadataFlag, extra...> &flags,
+                   bool take_union = false) {
```

Comment:
This does not get rid of the `std::initializer_list` overload but I think it does work.

I think there must be something special going on with initializer list that I don't understand, because I agree it should work based on my mental model of how this should work.

Reply:
Updated to the concept version but left the initializer list overload.

Reply:
Correction the linter doesn't like it. Reverting to the template template nonsense and making an issue, #839. Sorry.